woudc_data_registry/registry.py

```diff
@@ -80,44 +80,74 @@ def query_distinct(self, domain):
 
         return values
 
-    def query_by_field(self, obj, obj_instance, by):
+    def query_by_field(self, obj, by, value, case_insensitive=False):
         """
         query data by field
 
-        :param obj: object (field) to be queried
-        :param obj_instance: object instance to be queried
-        :param by: value to be queried
-
-        :returns: query results
+        :param obj: Class of table to query in
+        :param by: Field name to be queried
+        :param value: Value of the field in any query results
+        :param case_insensitive: Whether to query strings case-insensitively
+        :returns: Query results
         """
 
         field = getattr(obj, by)
-        value = getattr(obj_instance, by)
 
         LOGGER.debug('Querying for {}={}'.format(field, value))
-        results = self.session.query(obj).filter(field == value).all()
+        condition = func.lower(field) == value.lower() \
+            if case_insensitive \
+            else field == value
```

Comment:
In this special case, `field` is a SQLAlchemy object whose `__eq__` method returns some kind of SQL condition rather than a boolean. So using == works and is non-boolean.